config/peerpods/podvm/azure-podvm-image-handler.sh

```diff
@@ -774,9 +774,24 @@ function delete_image_using_id() {
     # IMAGE_ID shouldn't be empty
     [[ -z "${IMAGE_ID}" ]] && error_exit "IMAGE_ID is empty"
 
-    # Delete the image
+    # Rightmost element of the input is <image-version>
+    IMAGE_VERSION=${IMAGE_ID##*/}
+
+    # Get the id of the source image
+    SOURCE_ID=$(az sig image-version show --resource-group "${AZURE_RESOURCE_GROUP}" \
+        --gallery-name "${IMAGE_GALLERY_NAME}" \
+        --gallery-image-definition "${IMAGE_DEFINITION_NAME}" \
+        --gallery-image-version "${IMAGE_VERSION}" \
+        --query "storageProfile.source.id" --output tsv) ||
+        error_exit "Failed to get the source id"
+
+    # Delete the image version
     az sig image-version delete --ids "${IMAGE_ID}" ||
-        error_exit "Failed to delete the image"
+        error_exit "Failed to delete the image version"
+
```

Comment:
have you tried to `image delete`  without `image-version delete` i think it may delete also the version (i haven't tried)
LGTM overall

Reply:
> have you tried to `image delete` without `image-version delete` i think it may delete also the version (i haven't tried) LGTM overall

So I deployed kata and then I ran `az image delete`. Outcome is :
1. the image version still exists in the gallery
2. I can start a peer pod using this image version

(1) means that the deletion job should still do `az image-version delete` .
(2) seems to indicate that we might be able to `az image delete` the source image once the gallery is populated

One concern I have is that `az sig image-version show --query storageProfile.source.id` still shows the id of the source image that was just deleted. I don't know if this _stale_ id in the description could cause troubles. I'm thus inclined to keep the change as it is for now. WDYT @snir911 @bpradipt ?

Reply:
I just followed the existing pattern in this file... are you saying that the deletion path should not do `|| error_exit` ?

Reply:
That's fine, i meant this is how the job/script is configured to work (maybe not anymore)